ProjNet/CoordinateSystems/Projections/MapProjection.cs

```diff
@@ -264,79 +318,279 @@ public override string XML
+        /// Transforms a series of points from meters to unit of output coordinate. This is done by
+        /// adding <see cref="false_easting"/> or <see cref="false_northing"/> and
+        /// multiplying with <see cref="_reciprocalMetersPerUnit"/>
         /// </summary>
-        /// <param name="lonlat">The ordinates of the point</param>
-        /// <returns>The transformed ordinates</returns>
-        public double[] DegreesToMeters(double[] lonlat)
+        /// <param name="inXs">A series of x-ordinates</param>
+        /// <param name="inYs">A series of y-ordinates</param>
+        /// <param name="outXs">A buffer for x-ordinates</param>
+        /// <param name="outYs">A buffer for y-ordinates</param>
+        /// <param name="strideX">A stride value for x-ordinates</param>
+        /// <param name="strideY">A stride value for y-ordinates</param>
+        protected void MetersToTarget(ReadOnlySpan<double> inXs, ReadOnlySpan<double> inYs, Span<double> outXs, Span<double> outYs, int strideX, int strideY)
         {
-            // Convert to radians
-            var tmp = lonlat.Length == 2
-                          ? new[] {Degrees2Radians(lonlat[0]), Degrees2Radians(lonlat[1])}
-                          : new[] {Degrees2Radians(lonlat[0]), Degrees2Radians(lonlat[1]), Degrees2Radians(lonlat[2])};
+            for (int i = 0, j = 0; i < inXs.Length; i+=strideX, j += strideY)
+            {
+                outXs[i] = (inXs[i] + false_easting) * _reciprocalMetersPerUnit;
+                outYs[j] = (inYs[j] + false_northing) * _reciprocalMetersPerUnit;
+            }
+        }
 
-            // Convert to meters
-            var res = RadiansToMeters(tmp);
+#endregion
 
-            // Add false easting and northing, convert to unit
-            res[0] = (res[0] + false_easting)/_metersPerUnit;
-            res[1] = (res[1] + false_northing)/_metersPerUnit;
-            if (res.Length == 3) res[2] = res[2]/_metersPerUnit;
+#region Reverse methods
+
+        /// <summary>
+        /// Abstract method to convert a point (x, y, z) in meters to (lon, lat, z) in radians
+        /// </summary>
+        /// <param name="x">The x-ordinate</param>
+        /// <param name="y">The y-ordinate</param>
+        /// <param name="z">The z-ordinate</param>
+        /// <returns>Converted point.</returns>
+        protected abstract (double lon, double lat, double z) MetersToRadians(double x, double y, double z);
 
-            return res;
+        /// <summary>
+        /// Abstract method to convert a point from meters to radians
+        /// </summary>
+        /// <param name="x">The x-ordinate</param>
+        /// <param name="y">The y-ordinate</param>
+        /// <param name="z">The z-ordinate</param>
+        /// <returns>Converted point.</returns>
+        protected (double lon, double lat, double z) MetersToDegrees(double x, double y, double z)
+        {
+            (double lon, double lat, double _) = MetersToRadians(x, y, z);

```

Comment:
why discard resulting Z?

shouldn't it be more like:
```csharp
double lon, lat;
(lon, lat, z) = MetersToRadians(x, y, z);
```

Reply:
Because it is not touched. I have removed z from `MetersToRadians` and `RadiansToMeters` signature altogether.